### PR TITLE
Fix clusterqueue controller to update pending workloads

### LINKED ISSUE #148: Pending workloads not updated in ClusterQueue status at creation
**What happened**:

When workloads are created, we are not updating the corresponding CQ status.

**What you expected to happen**:

The status to be up to date

**How to reproduce it (as minimally and precisely as possible)**:

Repro: https://github.com/alculquicondor/kueue/commit/1ae8618cb687956bcc6fc361ea839490164faa29

**Anything else we need to know?**:

This is because we are only queueing the clusterqueue for admitted workloads in assignedWorkloadHandler https://github.com/kubernetes-sigs/kueue/blob/27b05145f94ac528ff92790efc70d62d8c9feafc/pkg/controller/core/clusterqueue_controller.go#L145

**Environment**:
- Kueue version (use `git describe --tags --dirty --always`): main

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

To keep a correct track of number of pending workloads in a ClusterQueue

#### Which issue(s) this PR fixes:

Fixes #148

### CODE REVIEW

**Review by ArangoGutierrez — APPROVED** [review]
Hold in case others want to review
/hold
/lgtm

File: pkg/controller/core/clusterqueue_controller.go [hunk 1]
```diff
@@ -140,52 +140,70 @@ func (r *ClusterQueue) Generic(e event.GenericEvent) bool {
...
+	oldReq := h.requestForWorkloadClusterQueue(oldW)
+	if oldReq != nil {
+		q.AddAfter(*oldReq, constants.UpdatesBatchPeriod)
 	}
 	newW := e.ObjectNew.(*kueue.QueuedWorkload)
-	if newW.Spec.Admission != nil && (oldW.Spec.Admission == nil || workloadFinished(newW) ||
-		newW.Spec.Admission.ClusterQueue != oldW.Spec.Admission.ClusterQueue) {
-		q.AddAfter(requestForWorkloadClusterQueue(newW), constants.UpdatesBatchPeriod)
+	newReq := h.requestForWorkloadClusterQueue(newW)
+	if newReq != nil && (oldReq == nil || oldReq.Name != newReq.Name) {
```
[the remark this answers is not in the document]
Comment (alculquicondor) [reply pkg/controller/core/clusterqueue_controller.go:166]: Done

**Pull request merged**